api/src/main/java/gov/cms/ab2d/api/controller/ErrorHandler.java

```diff
@@ -19,12 +21,41 @@
 @ControllerAdvice
 class ErrorHandler extends ResponseEntityExceptionHandler {
 
+    @Value("${api.retry-after.delay}")
+    private int retryAfterDelay;
+
+    // All errors that are not the fault of the client
     @ExceptionHandler(Exception.class)
-    @ResponseStatus(HttpStatus.BAD_REQUEST)
+    public ResponseEntity<JsonNode> serverException(final Exception e) throws IOException {
+        return generateFHIRError(e, HttpStatus.INTERNAL_SERVER_ERROR);
+    }
+
+    // Most errors that are the fault of the client
+    @ExceptionHandler(IllegalArgumentException.class)
     public ResponseEntity<JsonNode> assertionException(final Exception e) throws IOException {
+        return generateFHIRError(e, HttpStatus.BAD_REQUEST);
+    }
+
+    @ExceptionHandler(EntityNotFoundException.class)
```

Comment:
This is JPA-specific exception, which makes the service class a somewhat "leaky" abstraction, i.e. persistence-related stuff leaks to controllers. Let's just introduce our own exception class, just like you did with TooManyRequests: e.g. `ResourceNotFoundException`